src/app.rb

```diff
@@ -0,0 +1,121 @@
+require_relative './book'
+require_relative './rental'
+require_relative './student'
+require_relative './teacher'
+require_relative './classroom'
+require_relative './person'
+
+class App
+  # setters and getters
+  attr_accessor :books, :rentals, :people
+
+  # constructor method
+  def initialize
+    @books = []
+    @rentals = []
+    @people = []
+  end
+
+  # method for listing all books.
+  def list_all_books
+    if @books.empty?
+      puts 'No books available yet!, go ahead and create one'
+    else
+      @books.each_with_index do |book, index|
+        puts "#{index + 1} Title: #{book.title}, Author: #{book.author}"
+      end
+    end
+  end
+
+  # method for listing all people
+  def list_all_people
+    if @people.empty?
+      puts 'No one has been listed yet!, kindly add a person'
+    else
+      @people.each_with_index do |person, index|
+        puts "#{index + 1} name: #{person.name} age: #{person.age} person_id: #{person.id}"
+      end
+    end
+  end
+
+  # Create a person (teacher or student, not a plain Person).
+  def create_person()
+    puts 'Want to create a student(1) or a teacher(2)? [Input number]: '
+    opt = gets.chomp
+    case opt
+    when '1'
+      puts 'Age:'
+      age = gets.chomp
+      puts 'Name:'
+      name = gets.chomp
+      puts 'Has parental_permission? [Y/N]:'
+      parental_permission = gets.chomp
+      puts 'Classroom:'
+      classroom = gets.chomp
+      new_student = Student.new(age, name, parental_permission, classroom)
+      @people << new_student unless @people.include?(new_student)
+      puts "New student: #{name} aged: '#{age}' created successfully"
+    when '2'
+      puts 'Age:'
+      age = gets.chomp
+      puts 'Name:'
+      name = gets.chomp
+      puts 'Specialization'
+      specialization = gets.chomp
+      new_teacher = Teacher.new(age, name, specialization)
```

Comment:
- [ ] I was unable to create a teacher. Kindly arrange your arguments here

  ```
  new_teacher = Teacher.new(age, specialization, name)
  ```

- [ ] After fixing that you also fix your Teacher initialize method:

  ![image](https://user-images.githubusercontent.com/81468675/210255362-ac7cfc6d-5b01-4b75-8b50-746c04435f4f.png)

  You need to refactor to this so that your Person object is created well:

  ```
  def initialize(age, specialization, name = 'Unknown')
    @specialization = specialization
    super(age, name)
  end
  ```

Reply:
Change implemented